common/include/pcl/point_cloud.h

```diff
@@ -603,17 +606,17 @@ namespace pcl
 
     protected:
       // This is motivated by ROS integration. Users should not need to access mapping_.
-      std::shared_ptr<MsgFieldMap> mapping_;
+      [[deprecated("rewrite your code to avoid using it")]] boost::shared_ptr<MsgFieldMap> mapping_;
```

Comment:
Maybe reword the message to include what to do instead. Like `Use pcl::createMapping(vector<pcl::PCLPointField>, pcl::MsgFieldMap&)` instead

Reply:
I also had this thought but was not sure since it's not a true replacement. But since you also think this way, I'll update the message.

Reply:
It's not public, but since there was a public usage, people (maintainers) who compile from source (and check warnings 😅 ) will know

Reply:
I like the new sassy one 🤣

Reply:
Well yeah, sounds a bit presumptuous, but what else can I say :D Here it's not about how to create a mapping, but rather about not using PCL's protected space to store your stuff.